_studio/shared/include/libmfx_core_interface.h

```diff
@@ -170,20 +171,22 @@ class EncodeHWCaps
     template <class CAPS>
     mfxStatus SetHWCaps(GUID encode_guid, CAPS *hwCaps, mfxU32 array_size = 1)
     {
+        CAPS* dst = reinterpret_cast<CAPS*>(m_caps);
         if (!m_caps)
         {
             m_encode_guid = encode_guid;
             m_size = array_size;
             m_caps = malloc(sizeof(CAPS)*array_size);
             if (!m_caps)
                 return MFX_ERR_MEMORY_ALLOC;
-            memcpy_s(m_caps, sizeof(CAPS)*array_size, hwCaps, sizeof(CAPS)*array_size);
+
+            std::copy(hwCaps, hwCaps + array_size, dst);
```

Comment:
that's wrong. You assigned dst=m_caps before you actually allocated m_caps on line 179. Why you don't want to use reinterpret_cast<CAPS*>(m_caps) directly in std::copy?

Reply:
you are right. tried to not increase code line width too much, several reinterpret_casts occupy significant space and reduce readability. will fix